prober/icmp.go

```diff
@@ -243,13 +279,51 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 
 	rb := make([]byte, 65536)
 	deadline, _ := ctx.Deadline()
-	if err := socket.SetReadDeadline(deadline); err != nil {
+	if icmpConn != nil {
+		err = icmpConn.SetReadDeadline(deadline)
+	} else {
+		err = v4RawConn.SetReadDeadline(deadline)
+	}
+	if err != nil {
 		level.Error(logger).Log("msg", "Error setting socket deadline", "err", err)
 		return
 	}
 	level.Info(logger).Log("msg", "Waiting for reply packets")
 	for {
-		n, peer, err := socket.ReadFrom(rb)
+		var n int
+		var peer net.Addr
+		var err error
+		var hopLimit float64 = -1
```

Comment:
Could we use just this rather than also needing the bool?

Reply:
OK, I changed hopLimit to a global var. '-1' means "no hopLimit". I need to ignore cm info when cm is nil or hopLimit is -1 (SetControlMessage failed). cm could theoretically have a value whenever there is any oob data not necessary about hopLimit.

Reply:
> The unprivileged sockets should also be tested.

All Linux tests where already using unprivileged sockets. I changed the code to force using privileged sockets. And it works as expected. Windows always uses privilege sockets.